main.py

```diff
@@ -339,10 +350,8 @@ def evaluate_genomes(genomes, config):
                 print(f"{player.fail_reason:<20}")
                 # Set the player as failed
                 players_list[player_index].failed = True
-                # Remove them from the lists as we've modified the genome by reference (fucking hope I have)
-                players_list.pop(player_index)
-                models_list.pop(player_index)
-                genomes_list.pop(player_index)
+                # Remove them from the lists as we've modified the genome by reference (fucking hope I have) lol
+                remove_element_from_lists(player_index, players_list, models_list, genomes_list)

```

Comment:
Fancy shit. Yeah, I think the genome is modified by reference as it's never returned in any of the online doc examples I've seen. Might be worth checking this tho otherwise any improvements we may see could just be made by chance. However, me typing that, I realise that after each generation the NEAT library spits out the fitnesses of all genomes so it does pass the data by reference.

Reply:
Most things in python gets passed by reference so I think it should be the expected one.